### PR TITLE
Itp reader issues

### PR DESCRIPTION
**Pull request by fgrunewald**
This fixes a bunch of issues related to the itp parser:

- parse '#else' directives as mentioned in #362
- increase speed of parser by fixing #377 
- parse directives position_restraints, settles, dihedral_restraints
- increase speed of make_residue graph #379 

Open issues are:
- parsing cmap #257 
- nested ifdefs fail #284

### CODE REVIEW

**Comment by fgrunewald** [comment]
@pckroon can you give this an initial overview? Do you have any ideas how to deal with nested #ifdefs?

File: vermouth/gmx/itp_read.py [hunk 1]
```diff
@@ -118,14 +120,25 @@ def parse_pragma(self, line, lineno=0):
...
+        elif line.startswith("#else"):
+            if self.current_meta is None:
+               raise IOError("Your #ifdef section is orderd incorrectly."
+                             "At line {} I read #endif but I haven not read"
+                             "a ifdef before.".format(lineno))
+            else:
+               inverse = {"ifdef": "ifndef", "ifndef": "ifdef"}
+               tag = self.current_meta["tag"]
+               condition = inverse[self.current_meta["condition"]]
+               self.current_meta = {'tag': tag, 'condition': condition.replace("#", "")}
```
Comment (pckroon) [inline vermouth/gmx/itp_read.py:140]: indentlint. You could also loose the `else` entirely since the if raises.

**Review by pckroon — APPROVED** [review]
Tiny nitpick, but LGTM otherwise!

File: vermouth/tests/test_itp_files.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 Test that force field files are properly read.
 """
 
+from inspect import ismethod
```
Comment (pckroon) [inline vermouth/tests/test_itp_files.py:18]: Unused import

**Review by pckroon — APPROVED** [review]

File: vermouth/tests/test_itp_files.py (hunk 2 above)
Comment (fgrunewald) [reply vermouth/tests/test_itp_files.py:18]: ahh shoot yes that was my original idea for the last test. I'll remove it.

**Comment by fgrunewald** [comment]
@pckroon can you check the interactions in this list: https://manual.gromacs.org/documentation/current/reference-manual/topologies/topology-file-formats.html 

I think we are just missing pairs_nb and virtual_sites1. Since we're at it, I'd also add those and then be done with it.

**Comment by pckroon** [comment]
Seems that way. It would/might also be nice to (re)order the interactions. Either alphabetically or in the order that's in the GMX manual.

**Review by pckroon — APPROVED** [review]

**Pull request merged**